### PR TITLE
[BugFix] Fix histogram statistics convert date type error bug

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Because the date toString type used in BE is a literal value of time, such as "20200101". When FE calculates statistics, EpochSecond is used. The conversion needs to be done when getting the stats, otherwise it won't correspond.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 40 (67.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|25|37|67.57%|[51, 77, 105, 109, 113, 136, 137, 138, 139, 165, 166, 167]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java [hunk 1]
```diff
@@ -91,33 +99,77 @@ public List<TStatisticData> queryHistogramStatistics(long tableId, List<String>
...
         for (int i = 0; i < histogramObj.size(); ++i) {
             JsonArray bucketJsonArray = histogramObj.get(i).getAsJsonArray();
-            Bucket bucket = new Bucket(
-                    Double.parseDouble(bucketJsonArray.get(0).getAsString()),
-                    Double.parseDouble(bucketJsonArray.get(1).getAsString()),
+
+            double low;
+            double high;
+            if (type.isDate()) {
+                DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyyMMdd");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java:132]: dtf could be static

**Comment by kangkaisen** [comment]
This bug should be covered by daily test

**Comment by HangyuanLiu** [comment]
> This bug should be covered by daily test
OK. 
At present, there is unit test coverage. In the future, when test cases are added uniformly, I will cover this bug.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java [hunk 2]
```diff
@@ -91,33 +102,77 @@ public List<TStatisticData> queryHistogramStatistics(long tableId, List<String>
...
-        Map<Double, Long> mcv = new HashMap<>();
+        Map<Double, Long> topN = new HashMap<>();
         for (int i = 0; i < histogramObj.size(); ++i) {
             JsonArray bucketJsonArray = histogramObj.get(i).getAsJsonArray();
-            mcv.put(Double.parseDouble(bucketJsonArray.get(0).getAsString()),
-                    Long.parseLong(bucketJsonArray.get(1).getAsString()));
+
+            double key;
+            if (type.isDate()) {
+                DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyyMMdd");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java:165]: static

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 30 / 39 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|1|00.00%|[50]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|28|36|77.78%|[54, 80, 108, 112, 116, 140, 141, 168]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|2|2|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=9145&id=StarRocks_starrocks&open=AYI0nCBNgmW8QGajHn8T Remove this unnecessary cast to "double".

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**